datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -304,154 +304,75 @@ fn summarize_min_max(
     stat: &ParquetStatistics,
 ) {
     match stat {
-        ParquetStatistics::Boolean(s) => {
-            if let DataType::Boolean = fields[i].data_type() {
-                if s.has_min_max_set() {
-                    if let Some(max_value) = &mut max_values[i] {
-                        match max_value.update_batch(&[Arc::new(BooleanArray::from(
-                            vec![Some(*s.max())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                max_values[i] = None;
-                            }
-                        }
-                    }
-                    if let Some(min_value) = &mut min_values[i] {
-                        match min_value.update_batch(&[Arc::new(BooleanArray::from(
-                            vec![Some(*s.min())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                min_values[i] = None;
-                            }
-                        }
-                    }
-                    return;
-                }
+        ParquetStatistics::Boolean(s)
+            if DataType::Boolean == *fields[i].data_type() && s.has_min_max_set() =>
```

Comment:
Is it possible to pull out the `s.has_min_max_set()` check from the branches to a once off check at the start of the function? I think `ParquetStatistics` has an equivalent `has_min_max_set()` :thinking: